gatsby-plugin-webfonts/src/modules/selfHosted.js

```diff
@@ -0,0 +1,73 @@
+import postcss from "postcss";
+import postcssJs from "postcss-js";
+import path from "path";
+import { constants, copyFileSync, existsSync, mkdirSync } from "fs";
+
+const defaultFontOptions = {
+  fontDisplay: `swap`,
+};
+
+export default function selfHosted({ directory, pathPrefix = `` }, reporter) {
+  const getFontFace = async (font) => {
+    const { family, urls, ...cssProperties } = createFontOptions(font);
+
+    const src = Object.entries(urls).map(([format, url]) => {
+      const sourcePath = path.join(directory, url);
+      if (!existsSync(sourcePath)) {
+        reporter.panicOnBuild(
+          `Specified selfHosted font file missing: "${sourcePath}"`,
+        );
+        return ``;
+      }
+
+      const fileName = path.basename(url);
+      const cssDir = path.join(
+        pathPrefix ? pathPrefix : `/`,
+        `static`,
+        `webfonts`,
+      );
+      const outputDir = path.join(directory, `public`, cssDir);
```

Comment:
instead of outputdir you can use the cachefolder here
```
 // const outputDir = path.join(directory, `public`, cssDir);

      if (!existsSync(cacheFolder)) {
        mkdirSync(cacheFolder, { recursive: true }, (err) => {
          reporter.error(err);
        });
      }

      const outputPath = path.join(cacheFolder, fileName);
```

Reply:
and done.